src/components/graphs-panel.tsx

```diff
@@ -15,7 +15,7 @@ interface IGraphsPanelProps {
   preRecordings:SensorRecording[];
   predictionState: PredictionState;
   prediction: number[][];
-  onAddPrediction: (prediction:number[]) => void;
+  setPredictionF: (prediction:number[][]) => void;
```

Comment:
It might be more readable and less error prone in the future if you added something like:

```
type GraphData = number[][];
type Prediction = GraphData
```

and then used that for:

```
prediction: Prediction;
setPredictionF: (prediction:Prediction) => void;

...

setDataF: (data: GraphData) => void;
```